lib/Bio/EnsEMBL/DataCheck/Checks/MemberProductionCounts.pm

```diff
@@ -0,0 +1,219 @@
+  GROUPS         => ['compara', 'compara_protein_trees'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['CAFE_gene_family', 'family', 'gene_member', 'gene_member_hom_stats', 'gene_tree_root', 'genome_db']
+};
+
+sub tests {
+  my ($self) = @_;
+  my $dba = $self->dba;
+  my $dbc = $dba->dbc;
+  my $helper = $dbc->sql_helper;
+  my $gdb_adap = $dba->get_GenomeDBAdaptor;
+  
+  fk($dbc, "gene_member_hom_stats", "gene_member_id", "gene_member", "gene_member_id");
+  my $constraint = "tree_type = 'tree' AND ref_root_id IS NULL";
+  fk($dbc, "gene_member_hom_stats", "collection", "gene_tree_root", "clusterset_id", $constraint);
+
+  my $sqlCollections = qq/
+     SELECT DISTINCT clusterset_id 
+       FROM gene_tree_root 
+     WHERE $constraint
+   /;
+
+  my $collections = $helper->execute_simple( -SQL => $sqlCollections );
+
+  my @columns = qw(families gene_trees gene_gain_loss_trees orthologues paralogues homoeologues);
+
+  foreach my $collection ( @$collections ) {
+    my $sql_sums = qq/
+      SELECT SUM(families) AS sum_families, SUM(gene_trees) AS sum_gene_trees, SUM(gene_gain_loss_trees) AS sum_gene_gain_loss_trees, SUM(orthologues) AS sum_orthologues, SUM(paralogues) AS sum_paralogues, SUM(homoeologues) AS sum_homoeologues 
+        FROM gene_member_hom_stats 
+      WHERE collection = "$collection"
+    /;
+
+    my $sums = $helper->execute( -SQL => $sql_sums, -USE_HASHREFS => 1 );
+
+    #First check counts are > 0 for the collections from sql queries
+    my $sqlFamilies = q/
+      SELECT COUNT(*) 
+        FROM family
+    /; # To be tested only with default
+    my $sqlGeneTrees = qq/
+      SELECT COUNT(*) 
+        FROM gene_tree_root 
+      WHERE clusterset_id = "$collection"
+    /;
+    my $sqlPolyploids = q/
+      SELECT COUNT(*) 
+        FROM genome_db 
+      WHERE genome_component IS NOT NULL
+    /; # Assumes that the polyploid genomes are found in all the collections
+    my $sqlCAFETrees = qq/
+      SELECT COUNT(*) 
+        FROM gene_tree_root 
+          JOIN CAFE_gene_family 
+            ON gene_tree_root.root_id = gene_tree_root_id 
+      WHERE clusterset_id = "$collection"
+    /;
+
+    my $family_count = $helper->execute_single_result( -SQL => $sqlFamilies );
+    my $genetree_count = $helper->execute_single_result( -SQL => $sqlGeneTrees );
+    my $polyploid_count = $helper->execute_single_result( -SQL => $sqlPolyploids );
+    my $cafetrees_count = $helper->execute_single_result( -SQL => $sqlCAFETrees );
+
+    my @counts = ($family_count, $genetree_count, $cafetrees_count, $genetree_count, $genetree_count, $genetree_count, $polyploid_count);
+    # Ensure rows are non-empty for each collection where expected
+    if ( $collection eq 'default' && $dba->get_division() =~ /vertebrates/ ) {
+      my $desc_1 = "The number of entries in family are > 0 for the $collection collection";
+      cmp_ok( $family_count, ">", 0, $desc_1 );
```

Comment:
This line is checking that the family table is not empty. This DC is not the right place for that. There are other `cmp_ok` below that check the gene-trees and CAFE families. I guess these come from the `has*` variables in the HC. Those variables are not there to check that the content if those tables is correct, but to get the current state of those tables and set what we expect of the `gene_member_hom_stats` table